api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/ValueSetTranslatorImpl.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.translators.impl;
+
+import javax.annotation.Nonnull;
+
+import lombok.AccessLevel;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.hl7.fhir.r4.model.Enumerations;
+import org.hl7.fhir.r4.model.ValueSet;
+import org.openmrs.Concept;
+import org.openmrs.ConceptMap;
+import org.openmrs.ConceptReferenceTerm;
+import org.openmrs.ConceptSet;
+import org.openmrs.module.fhir2.api.FhirConceptSourceService;
+import org.openmrs.module.fhir2.api.translators.ValueSetTranslator;
+import org.openmrs.module.fhir2.model.FhirConceptSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+@Setter(AccessLevel.PACKAGE)
+public class ValueSetTranslatorImpl implements ValueSetTranslator {
+	
+	@Autowired
+	private FhirConceptSourceService conceptSourceService;
+	
+	@Override
+	public ValueSet toFhirResource(@Nonnull Concept concept) {
+		ValueSet valueSet = new ValueSet();
+		valueSet.setId(concept.getUuid());
+		valueSet.setTitle(concept.getName().getName());
+		valueSet.setStatus(Enumerations.PublicationStatus.ACTIVE);
+		
+		valueSet.setDate((concept.getDateChanged() != null) ? concept.getDateChanged() : concept.getDateCreated());
+		
+		String description = (concept.getDescription() == null || concept.getDescription().getDescription() == null) ? ""
+		        : concept.getDescription().getDescription();
+		valueSet.setDescription(description);
+		
+		if (!concept.getSet()) {
+			return valueSet;
+		}
+		
+		ValueSet.ValueSetComposeComponent compose = new ValueSet.ValueSetComposeComponent();
+		
+		for (ConceptSet conceptSet : concept.getConceptSets()) {
+			Concept conceptSetMember = conceptSet.getConcept();
```

Comment:
There is one tricky spot here: what happens if the concept set member is itself a concept set? Here it seems the thing to do is to include a reference in the `compose.include.valueSet` instead of adding it as a normal concept. Sorry that should've been in the original ticket.